In [81]:
# DEFINING REQUIRED PACKAGES ---------------------------------------
import pandas as pd #importing pandas to use dataframes
import numpy as np

In [31]:
# READING THE DATA FROM GOOGLE SHEETS ---------------------------------------
googleSheetId = '17T3D-Xrv4ZO1ldBy6F-FR3mmHMt_W_H4FaDz1ItUuYQ'
workSheetName = 'data'
URL = 'https://docs.google.com/spreadsheets/d/{0}/gviz/tq?tqx=out:csv&sheet={1}'.format(
    googleSheetId, workSheetName)
df = pd.read_csv(URL)
df

Time (s)
0       8.16
1       8.16
2       8.12
3       8.14
4       8.14
5       8.10
6       8.12
7       8.17
8       8.14
9       8.16
10      8.18
11      8.09
12      8.18
13      8.21
14      8.16
15      8.10
16      8.12
17      8.16
18      8.18
19      8.12
20      8.21
21      8.18
22      8.17
23      8.14
24      8.18
25      8.06
26      8.16
27      8.24
28      8.10
29      8.13

## ----------------
## PROBLEM 4.13
## ----------------

## a)

In [83]:
# DETERMINING THE STANDARD DEVIATION
std = df['Time (s)'].std()
mean = df.mean()[0]
print("The standard deviation of the time measurement is: sigma = " + "{:.2f}".format(std) + " sec.")

The standard deviation of the time measurement is: sigma = 0.04 sec.


$$\boxed{\sigma = 0.04 \ s}$$

## b)

In [76]:
df.groupby(pd.cut(df['Time (s)'], np.arange(mean-std,mean+std,std))).count()

Time (s)
Time (s)                
(8.11, 8.149]          9
(8.149, 8.189]        13

In [92]:
perc_ins = ((9+13)/30)
ins_exp = 30*0.68
ins_real = 30*perc_ins
perc_out = 1 - perc_ins
out_exp = 30*0.32
out_real = 30*perc_out
print("Inside the range expected: " + "{:.1f}".format(ins_exp))
print("Inside the range real: " + "{:.0f}".format(ins_real))
print("Outside the range expected: " + "{:.1f}".format(out_exp))
print("Outside the range real: " + "{:.0f}".format(out_real))

Inside the range expected: 20.4
Inside the range real: 22
Outside the range expected: 9.6
Outside the range real: 8


## c)

In [78]:
df.groupby(pd.cut(df['Time (s)'], np.arange(mean-2*std,mean+2*std,2*std))).count()

Time (s)
Time (s)                
(8.071, 8.149]        13
(8.149, 8.228]        15

In [95]:
perc_ins = ((15+13)/30)
ins_exp = 30*0.95
ins_real = 30*perc_ins
perc_out = 1 - perc_ins
out_exp = 30*0.05
out_real = 30*perc_out
print("Inside the range expected: " + "{:.1f}".format(ins_exp))
print("Inside the range real: " + "{:.0f}".format(ins_real))
print("Outside the range expected: " + "{:.1f}".format(out_exp))
print("Outside the range real: " + "{:.0f}".format(out_real))

Inside the range expected: 28.5
Inside the range real: 28
Outside the range expected: 1.5
Outside the range real: 2


## ----------------
## PROBLEM 4.17
## ----------------

## a)

In [96]:
std_mean = std/np.sqrt(30)
mean, std_mean

(8.149333333333335, 0.007174881685649284)

$$\boxed{t = (8.149 \pm 0.007) \ s}$$

## b)

The data has only two significant figures, while the final answer has 3. This is because our uncertainty commands the number of significant figures of the final result, and for this case the standard deviation of the mean selected has a low uncertainty, which implies a higher number of figures for this case. 

## ----------------
## PROBLEM 4.18
## ----------------

## a)

We know that, by definition, the standard deviation of the mean is defined by:

$$\sigma_{\overline{u}} = \frac{\sigma_u}{\sqrt{N}}$$

For a final uncertainty of ± 3 m/s, the required number of measurements N is determined by:

$$N = \left (\frac{\sigma_u}{\sigma_{\overline{u}}}\right )^2$$

In [102]:
std = 10
std_mean = 3
N = (std/std_mean)**2
N

11.111111111111112

11 measurements will be enough to give the student a final uncertainty of ± 3 m/s

## b)

In [107]:
std_mean = 0.5
N = (std/std_mean)**2
N

400.0

400 measurements will be enough to give the student a final uncertainty of ± 0.5 m/s